assert/assertions.go

```diff
@@ -300,14 +300,25 @@ func messageFromMsgAndArgs(msgAndArgs ...interface{}) string {
 // basis on which the alignment occurs).
 func indentMessageLines(message string, longestLabelLen int) string {
 	outBuf := new(bytes.Buffer)
-
-	for i, scanner := 0, bufio.NewScanner(strings.NewReader(message)); scanner.Scan(); i++ {
-		// no need to align first line because it starts at the correct location (after the label)
-		if i != 0 {
-			// append alignLen+1 spaces to align with "{{longestLabel}}:" before adding tab
-			outBuf.WriteString("\n\t" + strings.Repeat(" ", longestLabelLen+1) + "\t")
+	msgScanner := bufio.NewScanner(strings.NewReader(message))
+
+	// a buffer is set manually to store the tokenization of the message string
+	// so that we can re-use the same buffer for each line of the message without
+	// any additional allocations. We set the maximum buffer length to 1 more
+	// than the length of the message to avoid exceeding the default
+	// MaxScanTokenSize while scanning lines. This can happen when there is a
+	// single very long line. Refer to issue #1525
+	msgScanner.Buffer(nil, len(message)+1)
```

Comment:
Is this safe? As we know from the assert.Len case this can contain a very large formatted slice. Having more than doubled the memory consumed by the large slice by creating a string containing the formatted version of it. Do we want to double the formatted version again? Plus all the intermediate buffers that were allocated on the heap by msgScanner before the next GC cycle?

We could at least pre-allocate the buffer for msgScanner rather than passing in `nil`. But I'd prefer that we either split or truncate lines that are unreasonably long, or we could just truncate the entire message to less than `bufio.MaxScanTokenSize`?

Reply:
> We could at least pre-allocate the buffer for msgScanner rather than passing in nil.

Agreed. I've made this change.

> But I'd prefer that we either split or truncate lines that are unreasonably long, or we could just truncate the entire message to less than `bufio.MaxScanTokenSize`?

If we decide to truncate, we could probably truncate based on the dimensions of the terminal? This makes more sense from a UX point of view.

Reply:
Hey, @arjunmahishi. I really think we should truncate rather than allocate unknown amounts of memory. It turns out `Equal` actually already uses a function to truncate its values. Sorry to steal the issue but I've opened #1646 with how I think it could be done. What do you think?

Reply:
Cool. Better to reuse the existing functionality and keep it consistent. Closing this now.